examples/components/CCFRAUD_VERTICAL/traininsilo/contributor.py

```diff
@@ -0,0 +1,404 @@
+        self._batch_count_global = 0
+        self._running_global_metrics = {
+            name: 0 for name in self._running_global_metrics
+        }
+
+    def get_step(self):
+        """Provide average value for every metric since last `reset_step` call"""
+        return {
+            f"{self._prefix}_{name}": value / self._batch_count_step
+            for name, value in self._running_step_metrics.items()
+        }
+
+    def get_global(self):
+        """Provide average value for every metric since last `reset_global` call"""
+        return {
+            f"{self._prefix}_{name}": value / self._batch_count_global
+            for name, value in self._running_global_metrics.items()
+        }
+
+
+class CCFraudTrainer:
+    def __init__(
+        self,
+        model_name,
+        global_rank,
+        global_size,
+        global_comm,
+        train_data_dir="./",
+        test_data_dir="./",
+        model_path=None,
+        lr=0.01,
+        epochs=1,
+        batch_size=10,
+        experiment_name="default-experiment",
+        iteration_name="default-iteration",
+    ):
+        """Credit Card Fraud Trainer trains simple model on the Fraud dataset.
+
+        Args:
+            model_name(str): Name of the model to use for training, options: SimpleLinear, SimpleLSTM, SimpleVAE
+            train_data_dir(str, optional): Training data directory path
+            test_data_dir(str, optional): Testing data directory path
+            lr (float, optional): Learning rate. Defaults to 0.01
+            epochs (int, optional): Epochs. Defaults to 1
+            batch_size (int, optional): DataLoader batch size. Defaults to 64.
+
+        Attributes:
+            model_: Model
+            criterion_: BCELoss loss
+            optimizer_: Stochastic gradient descent
+            train_dataset_: Training Dataset obj
+            train_loader_: Training DataLoader
+            test_dataset_: Testing Dataset obj
+            test_loader_: Testing DataLoader
+        """
+
+        # Training setup
+        self._lr = lr
+        self._epochs = epochs
+        self._batch_size = batch_size
+        self._experiment_name = experiment_name
+        self._iteration_name = iteration_name
+        self._global_rank = global_rank
+        self._global_size = global_size
+        self._global_comm = global_comm
+
+        self.device_ = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        logger.info(f"Using device: {self.device_}")
+
+        self.train_dataset_, self.test_dataset_, self._input_dim = self.load_dataset(
+            train_data_dir, test_data_dir, model_name
+        )
+        self.train_loader_ = DataLoader(
+            self.train_dataset_, batch_size=batch_size, shuffle=False
+        )
+        self.test_loader_ = DataLoader(
+            self.test_dataset_, batch_size=batch_size, shuffle=False
+        )
+
+        # Build model
+        self._model_name = model_name
+        if self._global_rank == 0:
```

Comment:
why do we have this test on global_rank here?

but in host.py there's no such test. aren't we expecting all contributor to use the bottom model?